selfdrive/controls/controlsd.py

```diff
@@ -198,14 +198,15 @@ def update_events(self, CS):
 
     if self.can_rcv_error or (not CS.canValid and self.sm.frame > 5 / DT_CTRL):
       self.events.add(EventName.canError)
-    if (self.sm['health'].safetyModel != self.CP.safetyModel and self.sm.frame > 2 / DT_CTRL) or \
-       self.mismatch_counter >= 200:
-      self.events.add(EventName.controlsMismatch)
-    if not self.sm.alive['plan'] and self.sm.alive['pathPlan']:
-      # only plan not being received: radar not communicating
-      self.events.add(EventName.radarCommIssue)
-    elif not self.sm.all_alive_and_valid():
-      self.events.add(EventName.commIssue)
+    if not SIMULATION:
+      if (self.sm['health'].safetyModel != self.CP.safetyModel and self.sm.frame > 2 / DT_CTRL) or \
+        self.mismatch_counter >= 200:
+        self.events.add(EventName.controlsMismatch)
+      if not self.sm.alive['plan'] and self.sm.alive['pathPlan']:
+        # only plan not being received: radar not communicating
+        self.events.add(EventName.radarCommIssue)
+      elif not self.sm.all_alive_and_valid():
```

Comment:
this `all_alive_and_valid()` call fails with `ubloxRaw` and `frontFrame`, triggering the communication issue. Both are being addressed later on about lines 230~240 to then prevent reading from them when on simulation.

Reply:
we can conditionally ignore whether frontFrame and ubloxRaw are alive based on the SIMULATION flag. it's a kw arg to SubMaster